src/platforms/ctxlink/WiFi_Server.c

```diff
@@ -641,6 +655,123 @@ bool is_ip_address_assigned(void)
 	return res;
 }
 
+//
+// Format the current connection data for display to user
+//
+//	SSID = 'name'
+//	RSSI = xx
+//	ip = xxx.xxx.xxx.xxx
+//
+void wifi_get_ip_address(char *buffer, uint32_t size)
+{
+	char local_buffer[64] = {0};
+	memset(buffer, 0x00, size);
+	if (g_wifi_connected) {
+		snprintf(local_buffer, sizeof(local_buffer), "SSID = %s\n", conn_info.acSSID);
+		strncpy(buffer, local_buffer, size);
+		snprintf(local_buffer, sizeof(local_buffer), "RSSI = %d\n", conn_info.s8RSSI);
+		strncat(buffer, local_buffer, size);
+		snprintf(local_buffer, sizeof(local_buffer), "IP = %d.%d.%d.%d\n", conn_info.au8IPAddr[0],
+			conn_info.au8IPAddr[1], conn_info.au8IPAddr[2], conn_info.au8IPAddr[3]);
+		strncat(buffer, local_buffer, size);
+	} else {
+		memcpy(buffer, "Not connected\n", strlen("Not connected\n"));
+	}
+}
+
+//
+// Wait for app_state to spin with timeout
+//
+void app_task_wait_spin(void)
+{
+	uint32_t wait_timeout = 2000U;
+	while (true) {
+		platform_tasks();
+		if (app_state == app_state_spin)
+			break;
+		platform_delay(1U);
+		if (wait_timeout-- == 0U)
+			break;
+	}
+}
+
+//
+// Using the passed arguments, attempt to connect to a Wi-Fi AP
+//
+void wifi_connect(size_t argc, const char **argv, char *buffer, uint32_t size)
+{
+	char ssid[64] = {0};
+	char pass_phrase[64] = {0};
+	char *output_buffer = ssid;
+	char *delimeter;
+	bool first_element = true;
+	memset(buffer, 0x00, size);
+	//
+	// Iterate over the arguments received to build the SSID and passphrase
+	//
+	// The BMF command line parser treats spaces as delimiters and both SSID
+	// and the passphrase may have embedded spaces. The SSID and passphrase
+	// should have a comma separator.
+	//
+	// The following loop concatenates each arg and adds back the space delimiter.
+	//
+	// It does this initially into the SSID name, up until an argument is found
+	// that contains a comma. The string before the comma is concatenated with
+	// the SSID and the string after the comma is used as the first element of
+	// the passphrase.
+	//
+	// The remaining arguments are then concatenated into the passphrase with
+	// an space added between them.
+	//
+	for (size_t loop = 1; loop < argc; loop++) {
+		delimeter = strchr(argv[loop], ',');
```

Comment:
As this variable is only used inside the loop, we suggest moving its definition to this line, allowing it to become a little more `const` - `char *const delimeter = strchr(argv[loop], ',');`

Reply:
Fixed as per suggestion.